hazelcast/src/test/java/com/hazelcast/util/FilteringClassLoader.java

```diff
@@ -96,6 +95,28 @@ public InputStream getResourceAsStream(String name) {
         return delegatingClassLoader.loadClass(name);
     }
 
+    public boolean checkResourceExcluded(String resourceName) {
+        if (resourceName == null) {
+            return true;
+        }
+        // transform resource path as class name so we can check against excluded packages
+        String resourceAsClassName = resourceName.replace('/', '.').replace(".class", "");
```

Comment:
I believe this method is on the fast path and now a lot of litter is being created to determine of something should be excluded. 

This can easily be solved by maintaining a cache of everything that has been excluded and only create the litter if something hasn't been checked yet.